functionary/prompt.py

```diff
@@ -31,147 +187,543 @@ def from_message(cls, message: Dict) -> EndToken:
         """
         role = message["role"]
         if role == "user":
-            return EndToken.user
+            return self.end_user
         elif role == "system":
-            return EndToken.system
+            return self.end_system
         elif role == "function":
-            return EndToken.function
+            return self.end_function
         else:  # role = assistant
-            if "function_call" in message and message["function_call"] is not None:
-                return EndToken.function_call
+            if message.get("function_call", None) is not None:
```

Comment:
I think no need, if we want to check the format, we should implement it in server.py or vll_server.py **Or we can implement another method inside PromptTemplate for checking format for request**

Reply:
Can we do the check in server.py or vllm_server.py?

Reply:
we shouldn't implement on server.py or vllm_server.py but in prompt_template because only prompt_template know which kinds of format it can support. 
If implment in server.py, we need to do like:
if template_prompt is V1:
....
if template_prompt is V2:
...
By the way, we can implement **validating input messages** later, maybe not in this PR, as this one is already huge

Reply:
Ok